### ISSUE TITLE
Trying MIST_VAD on another dataset

### ISSUE THREAD

**Issue by AndreiFlorescu1410** [fjchange/MIST_VAD#27]
Hello,

I'm a 2nd year master student at University of Bucharest AI program, and I want to use your approach on my own dataset but I can't find information on how the features from I3D and C3D are produced. Could you give me more insight into how to use your method on another datasets?

Thanks.

**Comment by xubiao-ui** [comment]
> 你好，
> 
> 我是布加勒斯特大学人工智能项目的二年级硕士生，我想在我自己的数据集上使用您的方法，但我找不到有关如何生成 I3D 和 C3D 功能的信息。您能否让我更深入地了解如何在其他数据集上使用您的方法？
> 
> 谢谢。
Hello, I also want to do this. Have you solved it, please?

**Comment by mnik17** [comment]
Hey @AndreiFlorescu1410 , @xubiao-ui ,

Did you manage to do it? I'm a bit unsure how to prepare my data correctly.

**Comment by fjchange** [comment]
For training on new datasets, some parameters may have to be tuned. The following is my experience, which may require you to make reasonable adjustments to the new dataset. I recommend you to use C3D with fewer cases to adjust.
1. In stage 1, the MIL generator needs to do L2 normalization for C3D features. In addition, the size of the sub-bag is related to the length of the abnormal event in the video. If the length of the abnormal event is short, the part_len for C3D should be set smaller.
2. It is necessary to obtain a certain degree of accuracy based on stage 1 to generate effective pseudo labels, because self-training is essentially a positive cycle. If stage1 does not get good results, it is recommended to tune the parameters first.
3. In Stage2, the backbone should be frozen for several epochs but train the classifier only before full finetuning. Then, the pseudo labels may introduce a class imbalance problem, the class weight should be tuned. Generally, you may need to increase the class weight of the anomaly class.

**Comment by mnik17** [comment]
Hi @fjchange,

thank you for the reply!

As I'm somewhat inexperienced in the field I wanted to ask you if the testing features used in stage 1 training should actually be from the testing set or from a validation one? Additionally I would like to know what the .npy files in the test_mask_dir are supposed to contain? I know they are provided in the Baidu link, but when I try to download them it is asking me to register and that's apparently impossible from my country. If you could provide an example that would be really helpful.

Thanks!

**Comment by fjchange** [comment]
As there is no validation set in the existing VAD dataset, we use the testing set directly. Yet you can split part of the training set as a validation set alternatively.

As for the test_mask_dir, it is the frame-wise annotation provided by the ShanghaiTech dataset, which is used only during testing. I uploaded them in `data/test_frame_mask/`

As for BaiduYun, it may not need to register before downloading. If you still fail to download them, you can use the features I provide in `OneDrive` directly. As for the UCF-Crime frame-wise h5py file, you can generate it from the raw UCF-Crime raw video dataset.
